Code/GraphMol/RGroupDecomposition/RGroupScore.cpp

```diff
@@ -76,16 +90,22 @@ double matchScore(const std::vector<size_t> &permutation,
         unsigned int &count = matchSetVect[i][smiles];
         ++count;
 #ifdef DEBUG
+          std::cerr << i << " smiles:" << smiles << " " << count << std::endl;
         std::cerr << " Linker Score: "
                   << linkerMatchSet[rg->second->attachments] << std::endl;
 #endif
         ++i;
       }
     }
-
+    
     double tempScore = 0.;
-    for (const auto &matchSet : matchSetVect) {
+    const std::string no_rgroup;
```

Comment:
Why is this uninitialized?

Reply:
It is initialized, It's just using the default constructor.

Reply:
ok, let me rephrase: "Why are you using an empty key?"

Reply:
I'll make it more explicit if you like.

Reply:
I'm not saying that it's wrong, it's just unusual enough that it would be worth adding a comment.

Reply:
Happy to change it, but here are the default initialization rules
```
> What is default initialization?

a. For a POD class type or a built-in POD type like int, initialise to an indeterminate value (do nothing).
b. For a non-POD class type, call a default constructor which provides the initialised value.
c. For an array type, default initialise every element of the array. 
```

Reply:
I believe this was done partly to solve the "vexing parse" issue

```
std::string Foo();
```

Is that a variable or a function definition?

Reply:
My suggestion was that it's worth adding a comment to the code to explain that you are intentionally using an empty string as a key in the map.

Reply:
I did a small refactor and changed the name to EMPTY_RGROUP